bigchaindb/elections/election.py

```diff
@@ -186,49 +191,55 @@ def get_commited_votes(self, bigchain, election_pk=None):
                                                                   election_pk))
         return self.count_votes(election_pk, txns, dict.get)
 
-    def has_concluded(self, bigchain, current_votes=[], height=None):
+    def has_concluded(self, bigchain, current_votes=[]):
         """Check if the election can be concluded or not.
 
-        * Elections can only be concluded if the current validator set
-          is exactly equal to the validator set encoded in the election outputs.
+        * Elections can only be concluded if the validator set has not changed
+          since the election was initiated.
         * Elections can be concluded only if the current votes form a supermajority.
 
         Custom elections may override this function and introduce additional checks.
         """
+        if self.has_validator_set_changed(bigchain):
+            return False
 
         election_pk = self.to_public_key(self.id)
         votes_committed = self.get_commited_votes(bigchain, election_pk)
         votes_current = self.count_votes(election_pk, current_votes)
-        current_validators = self.get_validators(bigchain, height)
 
-        if self.is_same_topology(current_validators, self.outputs):
-            total_votes = sum(current_validators.values())
-            if (votes_committed < (2/3) * total_votes) and \
-                    (votes_committed + votes_current >= (2/3)*total_votes):
-                return True
+        current_validators = self.get_validators(bigchain)
+        total_votes = sum(current_validators.values())
```

Comment:
It occured to me that one could sum the `"amount"` of the outputs of the election to get the `total_votes` and avoid the query.